builtin/credential/aws/path_image_tag.go

```diff
@@ -0,0 +1,299 @@
+
+	// Fetch the image entry corresponding to the AMI name
+	imageEntry, err := awsImage(req.Storage, imageID)
+	if err != nil {
+		return nil, err
+	}
+	if imageEntry == nil {
+		return logical.ErrorResponse("image entry not found"), nil
+	}
+
+	// If RoleTag is empty, disallow creation of tag.
+	if imageEntry.RoleTag == "" {
+		return logical.ErrorResponse("tag creation is not enabled for this image"), nil
+	}
+
+	// Create a random nonce
+	nonce, err := createRoleTagNonce()
+	if err != nil {
+		return nil, err
+	}
+
+	// max_ttl for the role tag should be less than the max_ttl set on the image.
+	maxTTL := time.Duration(data.Get("max_ttl").(int)) * time.Second
+
+	// max_ttl on the tag should not be greater than the system view's max_ttl value.
+	if maxTTL > b.System().MaxLeaseTTL() {
+		return logical.ErrorResponse(fmt.Sprintf("Registered AMI does not have a max_ttl set. So, the given TTL of %d seconds should be less than the max_ttl set for the corresponding backend mount of %d seconds.", maxTTL/time.Second, b.System().MaxLeaseTTL()/time.Second)), nil
+	}
+
+	// If max_ttl is set for the image, check the bounds for tag's max_ttl value using that.
+	if imageEntry.MaxTTL != time.Duration(0) && maxTTL > imageEntry.MaxTTL {
+		return logical.ErrorResponse(fmt.Sprintf("Given TTL of %d seconds greater than the max_ttl set for the corresponding image of %d seconds", maxTTL/time.Second, imageEntry.MaxTTL/time.Second)), nil
+	}
+
+	if maxTTL < time.Duration(0) {
+		return logical.ErrorResponse("max_ttl cannot be negative"), nil
+	}
+
+	// Attach version, nonce, policies and maxTTL to the role tag value.
+	rTagValue, err := prepareRoleTagPlainValue(&roleTag{Version: roleTagVersion,
+		ImageID:  imageID,
+		Nonce:    nonce,
+		Policies: policies,
+		MaxTTL:   maxTTL,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the key used for creating the HMAC
+	key, err := hmacKey(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the HMAC of the value
+	hmacB64, err := createRoleTagHMACBase64(key, rTagValue)
+	if err != nil {
+		return nil, err
+	}
+
+	// attach the HMAC to the value
+	rTagValue = fmt.Sprintf("%s:%s", rTagValue, hmacB64)
+	if len(rTagValue) > 255 {
+		return nil, fmt.Errorf("role tag 'value' exceeding the limit of 255 characters")
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"tag_key":   imageEntry.RoleTag,
+			"tag_value": rTagValue,
+		},
+	}, nil
+}
+
+// verifyRoleTagValue rebuilds the role tag value without the HMAC,
+// computes the HMAC from it using the backend specific key and
+// compares it with the received HMAC.
+func verifyRoleTagValue(s logical.Storage, rTag *roleTag) (bool, error) {
+	// Fetch the plaintext part of role tag
+	rTagPlainText, err := prepareRoleTagPlainValue(rTag)
+	if err != nil {
+		return false, err
+	}
+
+	// Get the key used for creating the HMAC
+	key, err := hmacKey(s)
+	if err != nil {
+		return false, err
+	}
+
+	// TODO: for testing purposes. Remove this.
```

Comment:
@issacg As the PR notes above, this is still in testing and not ready for merging or production use. We opened it up for broader looks, but as Vishal noted there are still TODOs.